jib-core/src/integration-test/java/com/google/cloud/tools/jib/api/ContainerizerIntegrationTest.java

```diff
@@ -115,28 +115,24 @@ private static FileEntriesLayer makeLayerConfiguration(
 
   private static void assertDockerInspect(String imageReference)
       throws IOException, InterruptedException {
-    String dockerContainerConfig = new Command("docker", "inspect", imageReference).run();
+    String dockerInspectExposedPorts =
+        new Command("docker", "inspect", "-f", "'{{json .Config.ExposedPorts}}'", imageReference)
+            .run();
+    String dockerInspectLabels =
+        new Command("docker", "inspect", "-f", "'{{json .Config.Labels}}'", imageReference).run();
+    String dockerConfigEnv =
+        new Command("docker", "inspect", "-f", "{{.Config.Env}}", imageReference).run();
+    String history = new Command("docker", "history", imageReference).run();
+
     MatcherAssert.assertThat(
-        dockerContainerConfig,
+        dockerInspectExposedPorts,
         CoreMatchers.containsString(
-            "            \"ExposedPorts\": {\n"
-                + "                \"1000/tcp\": {},\n"
-                + "                \"2000/tcp\": {},\n"
-                + "                \"2001/tcp\": {},\n"
-                + "                \"2002/tcp\": {},\n"
-                + "                \"3000/udp\": {}"));
+            "\"1000/tcp\":{},\"2000/tcp\":{},\"2001/tcp\":{},\"2002/tcp\":{},\"3000/udp\":{}"));
     MatcherAssert.assertThat(
-        dockerContainerConfig,
-        CoreMatchers.containsString(
-            "            \"Labels\": {\n"
-                + "                \"key1\": \"value1\",\n"
-                + "                \"key2\": \"value2\"\n"
-                + "            }"));
-    String dockerConfigEnv =
-        new Command("docker", "inspect", "-f", "{{.Config.Env}}", imageReference).run();
+        dockerInspectLabels,
+        CoreMatchers.containsString("\"key1\":\"value1\",\"key2\":\"value2\""));
```

Comment:
(Not too familiar with Jib or these ITs) If the whitespace changes were causing the mismatch for String, would it make sense (or be possible) to convert this to something like a JsonObject and match JSON contents instead of String comparisons?

Reply:
That's a good suggestion! Jib uses jackson instead of gson, but there's probably a similar way to do JSON comparison. I'll merge this PR to unblock CI for now, and keep an issue open to follow-up on the test refactoring.